### ISSUE TITLE
Rule Change: No prefer-destructuring for array -> array assignments

### ISSUE THREAD

**Issue by asportnoy** [eslint/eslint#16144]
### What rule do you want to change?

prefer-destructuring

### What change to do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new option

### Example code

```js
let a = [1, 2];
let b = [3, 4];

// Using "normal" array assignments
a[1] = b[1];

// Using destructuring
[, a[1]] = b;
```

### What does the rule currently do for this code?

Enforces use of the second method above, which is hard to read.

### What will the rule do after it's changed?

Allow assigning array items to array items without destructuring.

### Participation

- [ ] I am willing to submit a pull request to implement this change.

### Additional comments

There may be other cases that bring up similar issues of readability but this is just the one I experienced.
